src/Uno.UWP/Collections/MaterializableList.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Uno.Collections
+{
+	/// <summary>
+	/// List using a materialized version to enumerate itself to prevent
+	/// "Collection was modified" error.
+	/// </summary>
+	/// <remarks>
+	/// THIS IS NOT THREAD-SAFE. It is designed to be used on
+	/// the UI thread.
+	/// </remarks>
+	public class MaterializableList<T> : IList<T>
+	{
+		private readonly List<T> _listImplementation = new List<T>();
+
+		public IEnumerator<T> GetEnumerator() => Materialized.GetEnumerator();
+
+		IEnumerator IEnumerable.GetEnumerator() => Materialized.GetEnumerator();
+
+		public void Add(T item)
+		{
+			_listImplementation.Add(item);
+			_materialized = null;
+		}
+
+		public void Clear()
+		{
+			_listImplementation.Clear();
+			_materialized = null;
+		}
+
+		public bool Contains(T item) => _listImplementation.Contains(item);
+
+		public void CopyTo(T[] array, int arrayIndex) => _listImplementation.CopyTo(array, arrayIndex);
+
+		public bool Remove(T item)
+		{
+			_materialized = null;
+			return _listImplementation.Remove(item);
```

Comment:
You could keep `_materialized` if `_listImplementation.Remove(item)` returns `false`